ejb/src/test/java/biz/karms/sinkit/ejb/impl/BlacklistCacheServiceEJBTest.java

```diff
@@ -0,0 +1,222 @@
+package biz.karms.sinkit.ejb.impl;
+
+import biz.karms.sinkit.ejb.cache.pojo.BlacklistedRecord;
+import biz.karms.sinkit.ioc.*;
+import com.google.gson.*;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Tomas Kozel
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class BlacklistCacheServiceEJBTest {
+
+    IoCRecord ioc1;
+    IoCRecord ioc2;
+    String key;
+    GsonBuilder gsonBuilder;
+    @Mock
+    private Logger log;
+
+    @Mock
+     RemoteCache<String, BlacklistedRecord> blacklistCache;
+
+    @InjectMocks
+    private BlacklistCacheServiceEJB blacklistCacheService;
+
+    @Before
+    public void setUp () throws Exception{
+        //ioc1, ioc2 and their corresponding key (both have the same)
+        //will be used throughout
+       ioc1 = new GsonBuilder().setDateFormat(IoCRecord.DATE_FORMAT)
+                .create().fromJson(new InputStreamReader(new FileInputStream("src/test/resources/ioc1.json")),
```

Comment:
@hugeox 
 * Are you sure these are not leaving dangling file descriptors? How about try-with-resources construct?
 * Isn't the path ```src/test....``` depending exactly on the place you run Maven from? Should you use resources via get resources and having it properly packaged by Maven?

Reply:
The path is relative to the pom which is used to run the tests so it is ok.

Reply:
@hugeox You cannot rely on that. You don't know that. It works just by accident in your particular setup. Use the get classlaoder and get resource as stream.